### ISSUE TITLE
Build error with RN 0.73

### ISSUE THREAD

**Issue by johnf** [stripe/stripe-react-native#1578]
**Describe the bug**
After upgrading to React Native 0.73 I saw the following build error.

```
> Task :stripe_stripe-react-native:compileDebugKotlin FAILED
e: file:///home/johnf/work/gladly/mobile/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/CardFormView.kt:31:70 Unresolved reference: style
```

**To Reproduce**
1. Create a new 0.73 RN app
2. Follow the stripe-react-native setup instructions
2. yarn android

**Expected behavior**
No Compile errors

**Additional context**

The code in question is
```kotlin
class CardFormView(context: ThemedReactContext) : FrameLayout(context) {
  private var cardForm: CardFormView = CardFormView(context, null, R.style.StripeCardFormView_Borderless)
  private var mEventDispatcher: EventDispatcher? = context.getNativeModule(UIManagerModule::class.java)?.eventDispatcher
  private var dangerouslyGetFullCardDetails: Boolean = false
  private var currentFocusedField: String? = null
```

I can't find ```style``` defined in any of the resource files. Not sure why it wasn't a compile error before RN 0.73.0

My quick fix is to apply the following diff

```diff
diff --git a/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/CardFormView.kt b/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/CardFormView.kt
index b78d457..6e485ce 100644
--- a/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/CardFormView.kt
+++ b/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/CardFormView.kt
@@ -28,7 +28,7 @@ import com.stripe.android.view.CardFormView
 import com.stripe.android.view.CardInputListener
 
 class CardFormView(context: ThemedReactContext) : FrameLayout(context) {
-  private var cardForm: CardFormView = CardFormView(context, null, R.style.StripeCardFormView_Borderless)
+  private var cardForm: CardFormView = CardFormView(context)
   private var mEventDispatcher: EventDispatcher? = context.getNativeModule(UIManagerModule::class.java)?.eventDispatcher
   private var dangerouslyGetFullCardDetails: Boolean = false
   private var currentFocusedField: String? = null
```

**Comment by mihailo-misic** [comment]
Hi @johnf I'm also having this issue right now.

The fix you're suggesting is unfortunately not viable for me.

These are the versions of the relevant modules that I'm using:
- react-native: `0.73.0`
- @stripe/stripe-react-native: `0.35.0`
- Kotlin: `1.8.22` (also tried: `1.8.20` and `1.8.0`)

**Comment by jgillick** [comment]
Is it attempting to reference this style from the Stripe Android SDK?
https://github.com/stripe/stripe-android/blob/e4bbf255cc5cba06949879638d8a5fa698f6144b/payments-core/res/values/styles.xml#L52-L54

**Comment by jgillick** [comment]
I believe this will fix it correctly (taken from the [Flutter fix](https://github.com/flutter-stripe/flutter_stripe/pull/1479/files#diff-3c5f88b974a7b077f26cc3fb96e7dbd365e57592e832917db00641045b9b1273) of the same bug)

```diff
diff --git a/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/CardFormView.kt b/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/CardFormView.kt
index b78d457..3ee8e2b 100644
--- a/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/CardFormView.kt
+++ b/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/CardFormView.kt
@@ -28,7 +28,7 @@ import com.stripe.android.view.CardFormView
 import com.stripe.android.view.CardInputListener

 class CardFormView(context: ThemedReactContext) : FrameLayout(context) {
-  private var cardForm: CardFormView = CardFormView(context, null, R.style.StripeCardFormView_Borderless)
+  private var cardForm: CardFormView = CardFormView(context, null, com.stripe.android.R.style.StripeCardFormView_Borderless)
   private var mEventDispatcher: EventDispatcher? = context.getNativeModule(UIManagerModule::class.java)?.eventDispatcher
   private var dangerouslyGetFullCardDetails: Boolean = false
   private var currentFocusedField: String? = null

```

CC @mihailo-misic

**Comment by TrustyTechSG** [comment]
Im also having this issue, after upgrade to 0.73 facing many "xxx" was not declared in function "xxx" build errors from Stripe.

**Comment by jgillick** [comment]
@TrustyTechSG To get a resolution, you'll need to share the exact build errors you're seeing.

**Comment by TrustyTechSG** [comment]
> @TrustyTechSG To get a resolution, you'll need to share the exact build errors you're seeing.

Sorry for not providing the logs

```
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:237507:7: warning: the variable "_scrollToPaper" was not declared in function "scrollToPaper"
      _scrollToPaper(viewTag, x, y, animated);
      ^~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:237555:7: warning: the variable "_setGestureState" was not declared in function "setGestureStateNative"
      _setGestureState(handlerTag, newState);
      ^~~~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:237846:18: warning: the variable "message" was not declared in function "message"
          return message;
                 ^~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:237878:5: warning: the variable "jest" was not declared in function "beforeTest"
    jest.useFakeTimers();
    ^~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:240108:26: warning: the variable "structuredClone" was not declared in function "createAnimationWithExistingTransform"
      newAnimationData = structuredClone(_config.AnimationsData[animationName]);
                         ^~~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:252397:91: warning: the variable "value" was not declared in anonymous function " 8395#"
...e), n.bind(null, true)) : n(false, value);
                                      ^~~~~
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':stripe_stripe-react-native:compileReleaseKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
BUILD FAILED in 6m 1s
```

**Comment by divyanshkumarsharma** [comment]
When can we accept fix for RN Version 0.73 with compile SDK 34

**Comment by jgillick** [comment]
@TrustyTechSG, it looks like the error you're seeing is unrelated to this issue. You should probably open a new issue on this project.

**Comment by jgillick** [comment]
@divyanshkumarsharma The PR (https://github.com/stripe/stripe-react-native/pull/1579) is in review now. If you need the fix sooner, consider using [patch-package](https://www.npmjs.com/package/patch-package) and the patch I included in [my comment](https://github.com/stripe/stripe-react-native/issues/1578#issuecomment-1853123295) above. This should unblock you. That's what I'm doing in my project until this is released in the next version.

**Comment by TrustyTechSG** [comment]
@jgillick I just checked the Error log from Expo console again, the first error same as yours. Sorry for not sending the full log, just thought it was too long.

```
> Task :stripe_stripe-react-native:compileReleaseKotlin FAILED
e: file:///home/expo/workingdir/build/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/CardFormView.kt:31:70 Unresolved reference: style
```

**Comment by AliRezaBeigy** [comment]
would you mind release a version to fix this error

**Comment by rameesp** [comment]
[https://github.com/stripe/stripe-react-native/issues/1583#issue-2056720907](url)
I am getting this error on React Native version 0.73.0

**Comment by TrustyTechSG** [comment]
We can we release the new version including this fix, this is a blocking issue.

**Comment by jgillick** [comment]
@TrustyTechSG Until they release the fix, you can unblock your project by either using [patch-package](https://www.npmjs.com/package/patch-package) or installing the package using the git commit hash (`f1fe1252d27a1e551a195d0ce9796c4810cec0e9` in your `package.json` file.

**Comment by ahmadAlMezaal** [comment]
> ```diff
> private var cardForm: CardFormView = CardFormView(context, null, com.stripe.android.R.style.StripeCardFormView_Borderless)
> ```

This solved the issue for me! We're using react-native version `0.73.0`
I believe this is a temporary fix until it's fixed from Stripe

**Comment by falk-stefan** [comment]
Hello!

I'm using `react-native` v0.73.0 (reverted from v0.75.0) and `@stripe/stripe-react-native` v0.35.0 but I keep getting this compilation error.

@ahmadAlMezaal which version-combination did it for ya'?

**Comment by charliecruzan-stripe** [comment]
released in 0.35.1

**Comment by daniloeder** [comment]
I've solved this problem after removing this line from package.json dependencies:
"expo-application": "~5.9.1".
